docs/examples/plot_2D_basis_function.py

```diff
@@ -0,0 +1,154 @@
+# For the 2D case, we can approximate our response function as follows:
+# $$
+# f(x,y) \\approx \alpha_i \cdot a_i(x) + \beta_j \cdot b_j(y)
+# $$
+# Here, we simply add two basis objects, 'a_basis' and 'b_basis', together to get the additive basis.
+
+
+additive_basis = a_basis + b_basis
+
+# Evaluate the basis functions for the given trajectory.
+eval_basis = additive_basis.evaluate(x_coord, y_coord)
+
+print(f"Sum of two 1D splines with {eval_basis.shape[0]} "
+      f"basis element and {eval_basis.shape[1]} samples:\n"
+      f"\t- a_basis had {a_basis._n_basis_funcs} elements\n\t- b_basis had {b_basis._n_basis_funcs} elements.")
+
+# %%
+# Plot the Additive Basis Elements
+# --------------------------
+# To plot a 2D additive basis set, we need to evaluate the basis on a grid of points over the basis function domain.
+# This can be done using the `neurostatslib.basis.evaluate_on_grid` method.
+
+X, Y, Z = additive_basis.evaluate_on_grid(200, 200)
+
+fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+
+# Only plot the support
+Z[Z == 0] = np.nan
+ax.plot_surface(X, Y, Z[5], cmap="viridis", alpha=0.8)
+ax.plot_surface(X, Y, Z[25], cmap="inferno", alpha=0.8)
+plt.title(f"Additive basis with {eval_basis.shape[0]} elements")
+plt.show()
+
+# %%
+# Combining Basis Types
+# ---------------------
+# Basis objects of different types can be combined through multiplication or addition.
+# This feature is particularly beneficial when one of the axes represents a periodic variable
+# such as an angle. A practical example would be characterizing responses to position
+# in a linear maze and the LFP phase angle.
+
+c_basis = nsl.basis.RaisedCosineBasisLinear(n_basis_funcs=8)
+
+# Multiply or add basis of different types
+prod_basis = b_basis * c_basis
+add_basis = b_basis + c_basis
+
+print(f"Product of two 1D splines of different type with {eval_basis.shape[0]} "
+      f"basis element and {eval_basis.shape[1]} samples:\n"
+      f"\t- b_basis is of type {b_basis.__class__}\n\t- c_basis is of type {c_basis.__class__}")
+
+# %%
+# N-Dimensional Basis
+# -------------------
+# If you need to model an N-dimensional response function, you can combine
+# N 1D basis objects. However, be aware that the dimension of the evaluated basis function
+# will increase exponentially with the number of dimensions.
+# For example, evaluating a product of $N$ 1D bases with $T$ samples and $K$ basis element,
+# will output a $K^N \times T$ matrix.
+
+T = 10
+n_basis = 8
+
+a_basis = nsl.basis.RaisedCosineBasisLinear(n_basis_funcs=n_basis)
+b_basis = nsl.basis.RaisedCosineBasisLinear(n_basis_funcs=n_basis)
+c_basis = nsl.basis.RaisedCosineBasisLinear(n_basis_funcs=n_basis)
+
+prod_basis_3 = a_basis * b_basis * c_basis
+samples = np.linspace(0, 1, T)
+eval_basis = prod_basis_3.evaluate(samples, samples, samples)
+
+print(f"Product of three 1D splines results in {prod_basis_3._n_basis_funcs} "
+      f"basis elements.\nEvaluation output of shape {eval_basis.shape}")
+
```

Comment:
plot some views of these? has to be lower d slices, but still probably useful

Reply:
I added a plot of a couple of slices